src/t_string.c

```diff
@@ -85,16 +87,18 @@ void setGenericCommand(client *c, int flags, robj *key, robj *val, robj *expire,
         if (getGenericCommand(c) == C_ERR) return;
     }
 
-    if ((flags & OBJ_SET_NX && lookupKeyWrite(c->db,key) != NULL) ||
-        (flags & OBJ_SET_XX && lookupKeyWrite(c->db,key) == NULL))
+    keyFound = lookupKeyWrite(c->db,key);
```

Comment:
maybe we wanna avoid that lookup when the flags are not set?
although that won't really reduce any overhead, since lookup saves another lookup later...

Reply:
yes... that was Madelyn's idea... to always call the lookup outside the function instead of inside...

Reply:
I would expect that optimization to be a rather significant amount of refactoring, so I'm not sure it's worth trying to plan for it now.

Reply:
ok, i suppose that as long as the double lookup is fixed we're good.
if we need we can refactor it again in the future.